src/Compilers/CSharp/Portable/Binder/PatternExplainer.cs

```diff
@@ -0,0 +1,507 @@
+                            return testedType.ToDisplayString();
+                    }
+                }
+
+                return null;
+            }
+
+            // Handle the special case of a type test and a type evaluation.
+            string tryHandleTypeTestAndTypeEvaluation()
+            {
+                if (evaluations.Length == 1 && constraints.Length == 1 &&
+                    constraints[0] is (BoundDagTypeTest { Type: var constraintType }, true) &&
+                    evaluations[0] is BoundDagTypeEvaluation { Type: var evaluationType } te &&
+                    constraintType.Equals(evaluationType, TypeCompareKind.AllIgnoreOptions))
+                {
+                    var typedTemp = new BoundDagTemp(te.Syntax, te.Type, te);
+                    return SamplePatternForTemp(typedTemp, constraintMap, evaluationMap, requireExactType: true);
+                }
+
+                return null;
+            }
+
+            // Handle the special case of a null test and a type evaluation to unbox a nullable value type
+            string tryHandleUnboxNullableValueType()
+            {
+                if (evaluations.Length == 1 && constraints.Length == 1 &&
+                    constraints[0] is (BoundDagNonNullTest _, true) &&
+                    evaluations[0] is BoundDagTypeEvaluation { Type: var evaluationType } te &&
+                    input.Type.IsNullableType() && input.Type.GetNullableUnderlyingType().Equals(evaluationType, TypeCompareKind.AllIgnoreOptions))
+                {
+                    var typedTemp = new BoundDagTemp(te.Syntax, te.Type, te);
+                    var result = SamplePatternForTemp(typedTemp, constraintMap, evaluationMap, requireExactType: false);
+                    // We need a null check. If not included in the result, add it.
+                    return (result == "_") ? "not null" : result;
+                }
+
+                return null;
+            }
+
+            // Handle the special case of a tuple pattern
+            string tryHandleTuplePattern()
+            {
+                if (input.Type.IsTupleType &&
+                    constraints.IsEmpty &&
+                    evaluations.All(e => e is BoundDagFieldEvaluation { Field: var field } && field.IsTupleElement()))
+                {
+                    var elements = input.Type.TupleElements;
+                    int cardinality = elements.Length;
+                    var subpatterns = new ArrayBuilder<string>(cardinality);
+                    subpatterns.AddMany("_", cardinality);
+                    foreach (BoundDagFieldEvaluation e in evaluations)
+                    {
+                        var elementTemp = new BoundDagTemp(e.Syntax, e.Field.Type, e);
+                        var index = e.Field.TupleElementIndex;
+                        if (index < 0 || index >= cardinality)
+                            return null;
```

Comment:
>return null [](start = 28, length = 11)

Are we testing this case?

Reply:
No. It would just cause us not to produce a tuple pattern in the face of a bad tuple API declaration. We would use a property pattern instead with `Item1` etc.

---
In reply to: [443843360](https://github.com/dotnet/roslyn/pull/44702#discussion_r443843360) [](ancestors = 443843360)

Reply:
Added a test,

---
In reply to: [443845046](https://github.com/dotnet/roslyn/pull/44702#discussion_r443845046) [](ancestors = 443845046,443843360)